lib/cc/cli/init.rb

```diff
@@ -51,29 +57,17 @@ def available_configs
         all_paths.reject { |path| ['.', '..'].include?(File.basename(path)) }
       end
 
-      def engine_eligible?(engine)
-        !engine["community"] && engine["enable_regexps"].present? && has_files?(engine)
-      end
+      def engines_enabled?
+        return @engines_enabled if @engines_enabled
```

Comment:
The one issue with this memoization is that if `@engines_enabled` is false, it won't use the memoized value. I think you can get around that by returning `unless @engines_enabled.nil?`